flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/JobReconciler.java

```diff
@@ -101,12 +104,22 @@ public JobReconciler(
                     restoreFromLastSavepoint(flinkApp, effectiveConfig);
                 } else if (upgradeMode == UpgradeMode.LAST_STATE) {
                     final String savepointLocation =
-                            flinkApp.getStatus().getJobStatus().getSavepointLocation();
+                            flinkApp.getStatus()
+                                    .getJobStatus()
+                                    .getSavepointInfo()
+                                    .getLastSavepoint()
+                                    .getLocation();
                     // Upgrade mode changes from savepoint -> last-state
                     deployFlinkJob(
                             flinkApp, effectiveConfig, Optional.ofNullable(savepointLocation));
                 }
             }
+            updateForSpecReconciliationSuccess(flinkApp);
+        } else if (SavepointUtils.shouldTriggerSavepoint(flinkApp)) {
+            if (flinkApp.getStatus().getJobStatus().getSavepointInfo().getTriggerId() == null) {
```

Comment:
This check could also be part of `shouldTriggerSavepoint` to make it cleaner

Reply:
makes more sense, yeah